### PR TITLE
feat(compiler): Support $baseUrl in HTML attributes when loading templates.

### PR DESCRIPTION
**Pull request by alxhub**
Addresses #2384.

### CODE REVIEW

File: modules/angular2/src/render/dom/compiler/view_loader.ts [hunk 1]
```diff
@@ -100,6 +115,31 @@ export class ViewLoader {
...
+  private _substituteBaseUrl(element, baseUrl: string): void {
+    if (DOM.isElementNode(element)) {
+      var attrs = DOM.attributeMap(element);
+      MapWrapper.forEach(attrs, (v, k) => {
+        if (isPresent(v) && v.indexOf('$baseUrl') >= 0) {
+          DOM.setAttribute(element, k, StringWrapper.replaceAll(v, /\$baseUrl/g, baseUrl));
+        }
+      });
+    }
+    let children = DOM.childNodes(element);
```
Comment (vsavkin) [inline modules/angular2/src/render/dom/compiler/view_loader.ts]: Maybe we should have a test checking the substitution works for child nodes. I think the test you have checks it for the root node.

Comment (alxhub) [inline modules/angular2/src/render/dom/compiler/view_loader.ts]: Good call, I'll add that.

**Comment by adamduren** [comment]
@alxhub What was the result of this? It looks like it was closed but not merged.

**Comment by adamduren** [comment]
It seems without something similar to this it would be difficult to distribute components that use resources in the template.

**Pull request closed without merging**